### PR TITLE
PVC GC

### PR DESCRIPTION
**Pull request by cd1989**
**What this PR does / why we need it**:

After WorkflowRun terminated, we need to clean data on PVC.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @zhujian7 @supereagle 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/workflow/common/const.go [hunk 1]
```diff
@@ -74,3 +75,9 @@ const (
 	// ContainerStateInitialized represents container is Running or Stopped, not Init or Creating.
 	ContainerStateInitialized ContainerState = "Initialized"
 )
+
+// Constants for GC pod.
+const (
+	GCContainerName = "gc"
+	GCDataPath = "/workspace"
```
Comment (supereagle) [inline pkg/workflow/common/const.go]: Format

File: pkg/workflow/controller/handlers/pod/gc.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	return true
+}
+
+// GCPodUpdated handles GC pod update. If GC pod is terminated, it will be deleted.
+func GCPodUpdated(client clientset.Interface, pod *corev1.Pod) {
+	if pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed {
+		if err := client.CoreV1().Pods(pod.Namespace).Delete(pod.Name, &metav1.DeleteOptions{}); err != nil {
+			if r, ok := err.(*errors.StatusError); ok && r.Status().Reason == metav1.StatusReasonNotFound {
```
Comment (supereagle) [inline pkg/workflow/controller/handlers/pod/gc.go]: Directly use `errors.IsNotFound(err)`?

Comment (cd1989) [reply pkg/workflow/controller/handlers/pod/gc.go]: You are right.

**Pull request merged**